core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -263,18 +263,22 @@ private void execute0(ClientRequestContext ctx, RedirectContext redirectCtx,
                     newReqDuplicator(reqDuplicator, responseHeaders, requestHeaders,
                                      nextReqTarget.toString(), nextAuthority);
 
-            if (isCyclicRedirects(redirectCtx, nextReqTarget.toString(), newReqDuplicator.headers().method())) {
+            final boolean isCyclicRedirects =
+                    !redirectCtx.addRedirectSignature(nextReqTarget,
+                                                      newReqDuplicator.headers().method());
+            if (isCyclicRedirects) {
                 handleException(ctx, derivedCtx, reqDuplicator, responseFuture, response,
                                 CyclicRedirectsException.of(redirectCtx.originalUri(),
```

Comment:
Optional) What do you think of raising either `CyclicRedirectsException` or `TooManyRedirectsException` by `addRedirectSignature()` which may streamline the code?

Reply:
I will try it as it seems to eliminate some extra methods. 👍